src/Maestro/maestro-angular/src/app/model/version-details.ts

```diff
@@ -0,0 +1,362 @@
+import { Subscription, Asset, Build } from 'src/maestro-client/models';
+import { Observable, combineLatest, of } from 'rxjs';
+import { map, switchMap, shareReplay } from 'rxjs/operators';
+import { StatefulResult, statefulPipe } from 'src/stateful';
+import { BuildService } from 'src/app/services/build.service';
+import { WrappedError, Loading, statefulSwitchMap } from 'src/stateful/helpers';
+import { HttpHeaders, HttpClient } from '@angular/common/http';
+import { DependencyDetail } from './dependency-detail';
+
+export class VersionDetails {
```

Comment:
![image](https://user-images.githubusercontent.com/44816719/68500470-a63d7800-0210-11ea-817b-7f03091531fa.png)

@ChadNedzlek Where would you prefer that this be implemented? And am I correct that you're suggesting we add the dependency information to the database in some way instead of using the files in the repos?

Reply:
Nah, the file in the repo is what we decided on long ago, and it's too late to change that.

More that there is a lot of logic here and reading of file.  The "darclib" already knows all that stuff, so it would be best to use it to handle figuring out the dependency tree.  The server already has all that code.  Having to re-write it all in TypeScript seems wasteful.

Instead of having a "get me the dependencys.xml" API call and parsing them all in TypeScript, have a server method that returns the information we want to display.  None of the methods have return types specified, and there doesn't appear to be any UI either, so I can't really tell what this code is attempting to do. But the TypeScript should generally only be "call some server API to do the heavy lifting, and then render it".

Reply:
My understanding was that that's not an option for this scenario. I'll look into it again and see if I can make it work.

Reply:
@ChadNedzlek On the surface it looks straight forward, however it would require either refactoring/moving or re-implementing RemoteFactory and possibly other types I haven't found yet that aren't implemented in darclib (this is the first error of the onion). If I'd known when I first started that adding it to the server was an option I probably would have gone that route, however at this point it's a new part of the code for me and I'd estimate that it'll take me a week to redo the work.

I agree that if it works that it's a better design, but I need to check with @chcosta as far as the timetable and priorities.

Reply:
We shouldn't check in the wrong code because of sunk cost, that's just increasing our debt.

Reply:
I agree with this entire conversation.  I'm unaware of any business priority which precludes doing the right thing here.

Reply:
Excellent, I'll update the PKPI schedule and start in on getting the other version together.